src/core/pointcloud/qgspointcloudblockhandle.cpp

```diff
@@ -0,0 +1,63 @@
+/***************************************************************************
+                         qgspointcloudblockhandle.cpp
+                         --------------------
+    begin                : March 2021
+    copyright            : (C) 2021 by Belgacem Nedjima
+    email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgspointcloudblockhandle.h"
+
+#include "qgstiledownloadmanager.h"
+#include "qgseptdecoder.h"
+
+//
+// QgsPointCloudBlockHandle
+//
+
+///@cond PRIVATE
+
+QgsPointCloudBlockHandle::QgsPointCloudBlockHandle( const QString &dataType, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, QgsTileDownloadManagerReply *tileDownloadManagerReply )
+  : mDataType( dataType ), mAttributes( attributes ), mRequestedAttributes( requestedAttributes ), mTileDownloadManagetReply( tileDownloadManagerReply )
+{
+  connect( mTileDownloadManagetReply, &QgsTileDownloadManagerReply::finished, this, &QgsPointCloudBlockHandle::blockFinishedLoading );
+}
```

Comment:
hmm it does not seem that the mTileDownloadManagetReply object would get deleted...

Reply:
That's true, I fixed that.